### DISCUSSION TITLE
Pick multiple images to send at once

### DISCUSSION THREAD

**Discussion by jackie-maxx** [flyerhq/flutter_chat_ui#690] [Ideas & Requests]
Please input feature with image album.

**Comment by demchenkoalex** [comment]
Hello! Can you clarify please?

**Comment by jackie-maxx** [comment]
> Hello! Can you clarify please?

Hello, I see your package have send picture but it send with only picture. so i want the feature send with multiple pictures. thanks.

**Comment by demchenkoalex** [comment]
ah yes, this will be implemented a bit later (preview of an added image will be on the top of the input) and then you will be able to send image + text in a same message and pick multiple images as well - not sure yet if those will be grouped in a single message or I will just send multiple image messages separately.

**Comment by asok** [comment]
@demchenkoalex that sounds great. But is it possible to have some workaround to have in the v1? If yes, are you able to write down some pseudo code how to do it?

> not sure yet if those will be grouped in a single message or I will just send multiple image messages separately.

my 2 cents would that having multiple images in a single message might prove easier to handle with pagination. If you'd have 1 image per message then when you pull from the a backend server it might happen that you pull "first" page and that "first" page might not include all messages that were created in a single multi file upload.

**Reply by demchenkoalex** [reply]
> But is it possible to have some workaround to have in the v1? If yes, are you able to write down some pseudo code how to do it?

I'm afraid not possible with `v1` as it will require lots of changes and I am super focused on delivering `v2` now. Obviously nothing stopping you from forking and doing it yourself, but it may be difficult as code quality is not the best there.

> my 2 cents would that having multiple images in a single message might prove easier to handle with pagination

true, I am just afraid of how to handle multiple loading/thumbhash states and having a "grid" of images, how they align etc and then what to do if more than x images included, how to split between multiple messages etc. Also changing image message type to have array instead of single source. Although, I can possibly do different message type like `MultiImage` that behaves differently and has different API.
